src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -222,7 +223,10 @@ private static FactoryContext CreateFactoryContext(RequestDelegateFactoryOptions
         factoryContext.MethodCall = CreateMethodCall(methodInfo, targetExpression, arguments);
 
         // Add metadata provided by the delegate return type and parameter types next, this will be more specific than inferred metadata from above
-        AddTypeProvidedMetadata(methodInfo, factoryContext.Metadata, factoryContext.ServiceProvider);
+        AddTypeProvidedMetadata(methodInfo,
+            factoryContext.Metadata,
+            factoryContext.ServiceProvider,
+            CollectionsMarshal.AsSpan(factoryContext.PropertiesAsParameter));
```

Comment:
Could we simplify things slightly by passing in all the `ParameterInfo` together instead of forcing AddTypeProvidedMetadata to iterate two lists?

```suggestion
            AddTypeProvidedMetadata(factoryContext.Parameters);
```

It doesn't look like `factoryContext.PropertiesAsParameter` is read anywhere else, so we might as well throw all the `ParameterInfo`s in there including the ones from when we called  `var arguments = CreateArguments(methodInfo.GetParameters(), factoryContext);`. I know we were reevaluating `methodInfo.GetParameters()` before, but we might as well improve this.

Calling `AsSpan` is a cool trick in `Flatten` where the source could be either an array or a list, but I don't see the point of turning the `List<ParameterInfo>` into a `Span<ParameterInfo>` here.